transcom/mymove
# INT B-18911 gun safe backend

## [B-18911](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18911)

## Summary

This branch adds changes to the backend to allow the app to track whether or not a customer has an entitlement to ship a gun safe. Currently the govt. just want's a flag to denote whether this is true or false, no weight calculations/fields have been added.

### How to test

1. Get the **Order** ID of an order through the web tools, easiest way is to access the "Edit orders" screen of a move then hit F12, you should see a uuid near the bottom of the "Network" tab, you'll want the top-level ID from this JSON set.
2. Use Postman to access the GHC/Office API at {{baseUrl}}/counseling/orders/:orderID/allowances and fill in the order ID in the URL.
3. Use the Postman Interceptor or set your cookie values as needed, you'll also need to set a `X-CSRF-Token` value in the request headers to the value of the `masked_gorilla` cookie.
4. For the body, use { "gunSafe": true } , Send and make sure you get a 200 OK response.
5. Either get the entitlement ID from the web browser tools, or open DBeaver (or any database tool), go to the `orders` table and find your order with the UUID you noted, go to the entitlement_id column and click the link (in DBeaver, small box to the left of the actual value), to open the appropriate entitlement in the `entitlements` table.
6. Scroll to the end where you should see a "gun_safe" field, which should now be checked.
7. If you send another Postman request with a body of "{ gunSafe: false }", you should be able to refresh DBeaver and see the value is now false.
